src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/VisitNotificationEventRepository.kt

```diff
@@ -1,8 +1,23 @@
 package uk.gov.justice.digital.hmpps.visitscheduler.repository
 
 import org.springframework.data.jpa.repository.JpaRepository
+import org.springframework.data.jpa.repository.Query
 import org.springframework.stereotype.Repository
 import uk.gov.justice.digital.hmpps.visitscheduler.model.entity.notification.VisitNotificationEvent
+import uk.gov.justice.digital.hmpps.visitscheduler.service.NotificationEventType
 
 @Repository
-interface VisitNotificationEventRepository : JpaRepository<VisitNotificationEvent, Int>
+interface VisitNotificationEventRepository : JpaRepository<VisitNotificationEvent, Int> {
+
+  @Query(
+    "SELECT count(ve) > 0" +
+      " FROM visit_notification_event ve " +
+      " WHERE ve.create_timestamp BETWEEN NOW() - INTERVAL '1 MINUTE' AND NOW() " +
```

Comment:
what if we remove this 1 minute check altogether even if they are not recent duplicates?

Reply:
maybe this is needed as we can have a non association for a 3rd prisoner added.